lib/src/delegates/asset_grid_drag_selection_aggregator.dart

```diff
@@ -0,0 +1,188 @@
+  void onDragStart(
+    BuildContext context,
+    LongPressStartDetails details,
+    int index,
+    AssetEntity entity,
+  ) {
+    isInDragging = true;
+
+    final scrollableState = _checkScrollableStatePresent(context);
+    if (scrollableState == null) {
+      return;
+    }
+
+    _autoScroller = EdgeDraggingAutoScroller(
+      scrollableState,
+      velocityScalar: _kDefaultAutoScrollVelocityScalar,
+    );
+
+    initialSelectedIdx = index;
+
+    dragSelect = !delegate.provider.selectedAssets.contains(entity);
+  }
+
+  void onDragUpdate(
+    BuildContext context,
+    LongPressMoveUpdateDetails details,
+    double itemSize,
+    int gridCount,
+    double topPadding,
+  ) {
+    if (!isInDragging) {
+      return;
+    }
+
+    if (dragSelect &&
+        delegate.provider.selectedAssets.length ==
+            delegate.provider.maxAssets) {
+      return;
+    }
+
+    final scrollableState = _checkScrollableStatePresent(context);
+    if (scrollableState == null) {
+      return;
+    }
+
+    final column = _getDragPositionIndex(details.globalPosition.dx, itemSize);
+    final row = _getDragPositionIndex(
+      details.globalPosition.dy -
+          topPadding -
+          (View.of(context).viewPadding.top /
+              View.of(context).devicePixelRatio) +
+          scrollableState.position.pixels,
+      itemSize,
+    );
+
+    final currentDragIndex = row * gridCount + column;
+
+    List<AssetEntity> filteredAssetList = <AssetEntity>[];
+    // add asset
+    if (currentDragIndex < initialSelectedIdx) {
+      filteredAssetList = delegate.provider.currentAssets
+          .getRange(
+            currentDragIndex,
+            math.min(
+                initialSelectedIdx + 1, delegate.provider.currentAssets.length),
+          )
+          .toList()
+        ..reversed;
+    } else {
+      filteredAssetList = delegate.provider.currentAssets
+          .getRange(
+            initialSelectedIdx,
+            math.min(
+                currentDragIndex + 1, delegate.provider.currentAssets.length),
+          )
+          .toList();
+    }
+
+    for (final asset in filteredAssetList) {
+      delegate.selectAsset(context, asset, currentDragIndex, !dragSelect);
+    }
+
+    final bool stopAutoScroll =
+        (!dragSelect && delegate.provider.selectedAssets.isEmpty) ||
+            (dragSelect &&
+                delegate.provider.selectedAssets.length ==
+                    delegate.provider.maxAssets);
+
+    if (stopAutoScroll) {
+      _autoScroller?.stopAutoScroll();
+      _autoScroller = null;
+      return;
+    }
+
+    if (!dragSelect && delegate.provider.selectedAssets.isEmpty) {}
+
+    _autoScroller?.startAutoScrollIfNecessary(
+      Rect.fromLTWH(
+        (column + 1) * itemSize,
+        details.globalPosition.dy > MediaQuery.sizeOf(context).height * 0.8
+            ? (row + 1) * itemSize
+            : math.max(topPadding, details.globalPosition.dy),
+        itemSize,
+        itemSize,
+      ),
```

Comment:
Consider explicitly declaring it as a local variable and more comments about the idea when calculating it.